### DISCUSSION TITLE
PostgreSQL 2 Schemas

### DISCUSSION THREAD

**Discussion by bernhard-ma-adm** [Budibase/budibase#7420] [Help] [answered]
Hi all,
i like BudiBase a lot - thx for all your support!

i have 1 Postgre Database with 2 custom schemas (bivema, diklabu).
In bivema exists a table 'dozent'
`dozent_id| name`

in diklabu exists a table 'stundenplan'
`stundenplan_id | date`
and a table 'zw_dozent_stundenplan' with the foreign-Keys from the referrenced Tables
`id | dozent_id | stundenplan_id`

In Budibase i can only see 1 schema in one DB-Connection. With 2 Connections i cannot define a relationship in BB between the two schemas.

Is it possible, to
a) see more than one schema in one connection or
b) define a relationship in Budibase over two connections
Alternative:
c) if i define a view in schema diklabu with the table bivema.dozent, i cannot fetch these view, because there is no primary Key defined.

Or do you have other suggestions for me?
Thanks!!!

**Comment by andz-bb** [comment, marked as answer]
Hey @bernhard-mischokacademy, unfortunately I don't think either option that you mentioned is going to work in Budibase at the minute. Do you have any way to consolidate your tables under one schema?

For your alternative suggestion I am aware there is an open issue at the minute around adding support for views. If you want you can track any progress or workarounds for it here: #6216 Sorry I don't have a nice solution for you at this stage!

**Reply by Bernie1502** [reply]
Hi @andz-bb , thx for your fast reply!
Ok, that's (both) not so good...there are 2 schemas because of separation and this is by design, we want to keep it so.

I'll track the ongoings on the issue with the views - this would be an alternative for me!
Thanks!!!

**Comment by rg2011** [comment]
for future reference for anyone looking for this same use case, please see #12717
